### PR TITLE
feat: Add feature to show punycode in IDN domains

### PR DESCRIPTION
**Pull request by RKNF404**
Adds to flag to show punycode in IDN domains.
This is a mitigation for IDN homograph attacks.

### CODE REVIEW

File: patches/add-feature-to-show-puny-code.patch [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ 
+ namespace url_formatter {
+ 
+@@ -318,6 +319,10 @@ IDNConversionResult IDNToUnicodeWithAdjustmentsImpl(
+   GetTopLevelDomain(host, &top_level_domain, &top_level_domain_unicode);
+ 
+   IDNConversionResult result;
++  if (base::FeatureList::IsEnabled(url::kShowPunycodeDomains)) {
++    result.result = host16;
++    return result;
```
Comment (qoijjj) [inline patches/add-feature-to-show-puny-code.patch:19]: there's a bunch of logic below this. is there a reason why we skip it?

File: patches/add-feature-to-show-puny-code.patch [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
++#include "url/url_features.h"
+ 
+ namespace url_formatter {
+ 
+@@ -318,6 +319,10 @@ IDNConversionResult IDNToUnicodeWithAdjustmentsImpl(
+   GetTopLevelDomain(host, &top_level_domain, &top_level_domain_unicode);
+ 
+   IDNConversionResult result;
++  if (base::FeatureList::IsEnabled(url::kShowPunycodeDomains)) {
++    result.result = host16;
```
Comment (qoijjj) [inline patches/add-feature-to-show-puny-code.patch:18]: do you mind giving a screenshot of what this looks like on?

File: patches/add-feature-to-show-puny-code.patch [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+index 584e93ac72bd1..b89fea48914d6 100644
+--- a/url/url_features.cc
++++ b/url/url_features.cc
+@@ -7,6 +7,10 @@
+ 
+ namespace url {
+ 
++BASE_FEATURE(kShowPunycodeDomains,
++             "ShowPunycodeDomains",
++             base::FEATURE_DISABLED_BY_DEFAULT);
```
Comment (qoijjj) [inline patches/add-feature-to-show-puny-code.patch:34]: if this is a clear security benefit, is there a downside to enabling it by default?

Comment (qoijjj) [reply patches/add-feature-to-show-puny-code.patch:34]: ah, shame

makes sense though

**Review by RoyalOughtness — APPROVED** [review]

**Pull request merged**